src/config.rs

```diff
@@ -0,0 +1,40 @@
+use crate::state::State;
+use serde::Deserialize;
+use sqlx::postgres::PgPool;
+use std::sync::Arc;
+
+#[derive(Deserialize, Clone)]
+pub struct PostgresConfig {
+    host: String,
+    user: String,
+    password: String,
+    db: String,
+}
+
+impl PostgresConfig {
+    pub fn database_url(&self) -> String {
+        format!(
+            "postgres://{}:{}@{}/{}",
+            self.user, self.password, self.host, self.db
+        )
+    }
+}
```

Comment:
Is there a use on this struct beyond using these fields to build the`DATABASE_URL`?

Why not having a `database_url` field for the config `struct` and build this connection string straight from the environment variables when creating an instance of this config?

Reply:
Sure!